pg-extern-attr/src/lib.rs

```diff
@@ -285,11 +281,27 @@ fn impl_info_for_fn(item: &syn::Item) -> TokenStream {
                     result.into_datum()
                 }
                 Err(err) => {
+                    use std::sync::atomic::compiler_fence;
+                    use std::sync::atomic::Ordering;
+                    use pg_extend::error;
+
                     // ensure the return value is null
                     func_info.isnull = pg_extend::pg_bool::Bool::from(true).into();
 
-                    // TODO: anything else to cean up before resuming the panic?
-                    panic::resume_unwind(err)
+                    // The Rust code paniced, we need to recover to Postgres via a longjump
+                    //   A postgres logging error of Error will do this for us.
+                    compiler_fence(Ordering::SeqCst);
+                    let level = pg_extend::pg_error::Level::Error;
+
+                    if let Some(msg) = info.payload().downcast_ref::<&'static str>() {
+                        error!("panic executing Rust '{}': {}", stringify!(#func_name), msg);
```

Comment:
I'm realizing that this error is relying on the panic handler to catch the longjmp from C, which converts it to a panic, which then longjmps...

Which is very likely wrong... we may want to longjmp here directly.

Reply:
This appears to work as expected, so leaving.